filebeat/tests/system/test_modules.py

```diff
@@ -95,14 +97,21 @@ def run_on_file(self, module, fileset, test_file, cfgfile):
         objects = [o["_source"] for o in res["hits"]["hits"]]
         assert len(objects) > 0
         for obj in objects:
-            self.assert_fields_are_documented(obj)
-            # assert "error" not in obj  # no parsing errors
-            assert obj["fileset"]["module"] == module
+            assert obj["fileset"]["module"] == module, "expected fileset.module={} but got {}".format(
+                module, obj["fileset"]["module"])
+
+            if not (module == "mysql" and fileset == "slowlog") and not (module == "system" and fileset == "auth"):
+                # TODO: There are errors parsing the test logs from these modules.
+                assert "error" not in obj
+
+            if module != "system" and fileset != "audit":
+                # There are dynamic fields in audit logs that are not documented.
```

Comment:
It's possible, but I don't think using a sub-dict in this case provides any additional context to the user about the fields. We would be using sub-dict here due to the implementation details (like the fact that it uses a kv processor instead of a specific grok pattern for each audit record type). From a user perspective I don't think a sub-dict would add value (at least not with what I'm thinking -- `system.audit.kv.*`), but I don't like to be the one breaking precedent.